src/inference/src/dev/core_impl.cpp

```diff
@@ -207,6 +207,34 @@ static const auto core_properties_names =
 
 static const auto auto_batch_properties_names =
     ov::util::make_array(ov::auto_batch_timeout.name(), ov::hint::allow_auto_batching.name());
+
+ov::util::Path extract_weight_path(const std::string& compiled_properties) {
+    if (auto start = compiled_properties.find(ov::weights_path.name()); start != std::string::npos) {
+        start += std::string_view{ov::weights_path.name()}.size() + 1;
+        auto length = compiled_properties.find(",", start);
+        if (length != std::string::npos) {
+            length -= start;
+        }
+        return {compiled_properties.substr(start, length)};
+    } else {
+        return {};
+    }
+}
+
+ov::SoPtr<ov::ICompiledModel> import_compiled_model(const ov::Plugin& plugin,
+                                                    const ov::SoPtr<ov::IRemoteContext>& context,
+                                                    const ov::AnyMap& config) {
+    ov::SoPtr<ov::ICompiledModel> compiled_model;
+    if (config.count(ov::hint::compiled_blob.name())) {
+        try {
+            std::ifstream empty{};
+            compiled_model = context ? plugin.import_model(empty, context, config) : plugin.import_model(empty, config);
```

Comment:
I think you also need to pass weights_path / model_ptr to config to ensure that weightless compiled blob can also be imported.

Secondly, why do we pass empty fstream? I think you need to pass 
```c++
SharedStreamBuffer buf{reinterpret_cast<char*>(compiled_blob.data()), compiled_blob.get_byte_size()};
std::istream stream(&buf);
```
similarly to code in `read_cache_entry`

Reply:
It should not be required here but add it for compatibility.